src/pages/MarketPage/components/AllItemsSection.jsx

```diff
@@ -1,113 +1,108 @@
-import React, { useEffect, useState } from 'react'
-import { getProducts } from '../../../api/itemApi'
-import ItemCard from './ItemCard'
-import { ReactComponent as SortIcon } from '../../../assets/images/icons/ic_sort.svg'
-import { ReactComponent as SearchIcon } from '../../../assets/images/icons/ic_search.svg'
-import { Link } from 'react-router-dom'
-import DropdownList from '../../../components/UI/DropdownList'
-import PaginationBar from '../../../components/UI/PaginationBar'
+import React, { useEffect, useState } from "react";
+import { getProducts } from "../../../api/itemApi";
+import ItemCard from "./ItemCard";
+import { ReactComponent as SearchIcon } from "../../../assets/images/icons/ic_search.svg";
+import { Link } from "react-router-dom";
+import DropdownMenu from "../../../components/UI/DropdownMenu";
+import PaginationBar from "../../../components/UI/PaginationBar";
+import LoadingSpinner from "../../../components/UI/LoadingSpinner";
 
 const getPageSize = () => {
-  const width = window.innerWidth
+  const width = window.innerWidth;
   if (width < 768) {
     // Mobile viewport
-    return 4
+    return 4;
   } else if (width < 1280) {
     // Tablet viewport
-    return 6
+    return 6;
   } else {
     // Desktop viewport
-    return 10
+    return 10;
   }
-}
+};
 
-function AllItemsSection () {
-  const [orderBy, setOrderBy] = useState('recent')
-  const [page, setPage] = useState(1)
-  const [pageSize, setPageSize] = useState(getPageSize())
-  const [itemList, setItemList] = useState([])
-  const [isDropdownVisible, setIsDropdownVisible] = useState(false)
-  const [totalPageNum, setTotalPageNum] = useState()
+function AllItemsSection() {
+  const [orderBy, setOrderBy] = useState("recent");
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(getPageSize());
+  const [itemList, setItemList] = useState([]);
+  const [totalPageNum, setTotalPageNum] = useState();
+  const [isLoading, setIsLoading] = useState(true);
 
   const fetchSortedData = async ({ orderBy, page, pageSize }) => {
-    const products = await getProducts({ orderBy, page, pageSize })
-    setItemList(products.list)
-    setTotalPageNum(Math.ceil(products.totalCount / pageSize))
-  }
+    setIsLoading(true);
+    try {
+      const products = await getProducts({ orderBy, page, pageSize });
+      setItemList(products.list);
+      setTotalPageNum(Math.ceil(products.totalCount / pageSize));
```

Comment:
P3:
반환되는 데이터를 아래처럼 사용하실 수도 있습니다~

```suggestion
      const { list, totalCount } = await getProducts({ orderBy, page, pageSize });
      setItemList(list);
      setTotalPageNum(Math.ceil(totalCount / pageSize));
```